core/controlplane/config/templates/cloud-config-controller

```diff
@@ -1126,26 +1126,59 @@ write_files:
                     REGION=$(metadata dynamic/instance-identity/document | jq -r .region)
                     [ -n "${REGION}" ]
 
-                    while sleep 30; do
-                      STATE=$(asg describe-auto-scaling-instances --region "${REGION}" --instance-ids "${INSTANCE_ID}" | jq -r '.AutoScalingInstances[].LifecycleState')
-                      if [ ! "${STATE}" = Terminating:Wait ]; then
-                        continue
+                    # Not customizable (for now)
+                    SPOT_POLL_INTERVAL=5
+                    SG_POLL_INTERVAL=30
+
+                    # Used to control when to execute each termination check
+                    secs=0
+
+                    # Used to identify which was the source that requested the instance termination ('spot' or 'asg', for now)
+                    termination_source=''
+
+                    # Instance termination detection loop
+                    while sleep ${SPOT_POLL_INTERVAL}; do
```

Comment:
I created one spot instance to mess around with those `/latest/meta-data/spot/*` endpoints, but in my tests this `/latest/meta-data/spot/` returns 404, so I think there's no way of distinguishing between a spot and an on-demand/reserved instance at this level.